### PR TITLE
Battery input

### PR DESCRIPTION
**Pull request by rusefillc**
steps towards self-contained WBO unit

### CODE REVIEW

File: firmware/sampling.cpp [hunk 1]
```diff
@@ -94,3 +97,8 @@ float GetPumpNominalCurrent()
     constexpr float ratio = -1000 / (PUMP_CURRENT_SENSE_GAIN * LSU_SENSE_R);
     return pumpCurrentSenseVoltage * ratio;
 }
+
+float GetBatteryVoltage()
```
Comment (mck1117) [inline firmware/sampling.cpp:101]: this should either return the CAN-received battery voltage, or local battery voltage if present

File: firmware/sampling.cpp [hunk 2]
```diff
@@ -54,6 +55,8 @@ static void SamplingThread(void*)
             (1 - PUMP_FILTER_ALPHA) * pumpCurrentSenseVoltage +
             PUMP_FILTER_ALPHA * (result.PumpCurrentVoltage - result.VirtualGroundVoltageInt);
 
+        battv = result.BatteryVoltage;
```
Comment (mck1117) [inline firmware/sampling.cpp:58]: call `SetBatteryVoltage` instead here

File: firmware/sampling.cpp (hunk 1 above)
Comment (dron0gus) [reply firmware/sampling.cpp:101]: Idea was to apply some threshold logic to locally measured battery voltage.
See https://github.com/rusefi/wideband/commit/1f1792cd2987daa2157d353413823a33d105ef84

**Comment by mck1117** [comment]
hand merged in b838c414fa1b6ed44df4ab965ea68aca2759be17 ea262c382251918ea11c7535ee98f00343b06575 dc077caff19492a0e5a908410bfd44ba4c9d31bf

**Pull request closed without merging**